qt/RecentDocuments.cpp

```diff
@@ -148,5 +156,51 @@ namespace RecentDocuments
         LOG_DBG("RecentDocuments::getForAppType: returning " << recentDocs->size() << " valid documents");
         return recentDocs;
     }
+
+    Poco::JSON::Array::Ptr getForAllTypes()
+    {
+        std::vector<DocumentEntry> allDocs;
+
+        LibreOfficeKitDocumentType types[] = { LOK_DOCTYPE_TEXT, LOK_DOCTYPE_SPREADSHEET,
+                                               LOK_DOCTYPE_PRESENTATION };
+        for (LibreOfficeKitDocumentType type : types)
+        {
+            Poco::JSON::Array::Ptr typeDocs = getForAppType(type);
+
+            for (size_t i = 0; i < typeDocs->size(); ++i)
+            {
+                Poco::JSON::Object::Ptr docObj = typeDocs->getObject(i);
+                QString uri = QString::fromStdString(docObj->getValue<std::string>("uri"));
+                QString name = QString::fromStdString(docObj->getValue<std::string>("name"));
+                QString timestampStr =
+                    QString::fromStdString(docObj->getValue<std::string>("timestamp"));
+                QDateTime timestamp = QDateTime::fromString(timestampStr, Qt::ISODate);
+
+                allDocs.push_back({ uri, name, timestamp, type });
+            }
+        }
+
+        std::sort(allDocs.begin(), allDocs.end(),
+                  [](const DocumentEntry& a, const DocumentEntry& b) {
+                      return a.timestamp > b.timestamp;
```

Comment:
I'd use a predicate here that takes into account all four members of `DocumentEntry` (so that the non-stable order into which `std::sort` puts equivalent elements doesn't lead to random results)